File: lib/models/camp/camp_model.dart

import 'time_run_model.dart';

class CampModel {
  String campaignId;
  String campaignName;
  String status;
  String videoId;
  String fromDate;
  String toDate;
  String fromTime;
  String toTime;
  String daysOfWeek;
  String videoType;
  String urlYoutube;
  String urlUSP;
  String computerId;
  List<TimeRunModel>? lstTimeRun;

  CampModel({
    required this.campaignId,
    required this.campaignName,
    required this.status,
    required this.videoId,
    required this.fromDate,
    required this.toDate,
    required this.fromTime,
    required this.toTime,
    required this.daysOfWeek,
    required this.videoType,
    required this.urlYoutube,
    required this.urlUSP,
    required this.computerId,
    required this.lstTimeRun,
  });

  factory CampModel.fromJson(Map<String, dynamic> json) {
    return CampModel(
      campaignId: json['campaign_id'] ?? '',
      campaignName: json['campaign_name'] ?? '',
      status: json['status'] ?? '',
      videoId: json['video_id'] ?? '',
      fromDate: json['from_date'] ?? '',
      toDate: json['to_date'] ?? '',
      fromTime: json['from_time'] ?? '',
      toTime: json['to_time'] ?? '',
      daysOfWeek: json['days_of_week'] ?? '',
      videoType: json['video_type'] ?? '',
      urlYoutube: json['url_youtobe'] ?? '',
      urlUSP: json['url_usp'] ?? '',
      computerId: json['computer_id'] ?? '',
      lstTimeRun: json['lstTimeRun'] ?? [],
    );
  }
}

File: android/app/src/main/kotlin/com/example/play_box/model/camp/CampModel.kt
package com.example.play_box.model.camp

import com.google.gson.annotations.SerializedName

data class CampModel(
    @SerializedName("campaign_id")
    val campaignId: String,
    @SerializedName("campaign_name")
    val campaignName: String?,
    val status: String?,
    @SerializedName("video_id")
    val videoId: String?,
    @SerializedName("from_date")
    val fromDate: String?,
    @SerializedName("to_date")
    val toDate: String?,
    @SerializedName("from_time")
    val fromTime: String?,
    @SerializedName("to_time")
    val toTime: String?,
    @SerializedName("days_of_week")
    val daysOfWeek: String?,
    @SerializedName("video_type")
    val videoType: String?,
    @SerializedName("url_youtobe")
    val urlYoutube: String?,
    @SerializedName("url_usp")
    val urlUSP: String?,
    @SerializedName("computer_id")
    val computerId: String?,
    var listTimeRun: List<TimeRunModel>?
)

data class TimeRunModel(
    @SerializedName("id_run")
    val idRun: String?,
    @SerializedName("campaign_id")
    val campaignId: String?,
    @SerializedName("from_time")
    val fromTime: String,
    @SerializedName("to_time")
    val toTime: String
)